src/Conversion/ONNXToTOSA/Math/Elementwise.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------------- Elementwise.cpp - Elementwise Op --------------------===//
+//
+// Copyright (c) 2022 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file lowers ONNX element-wise operators to TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+namespace {
+
+template <typename ONNXOpT, typename TOSAOpT>
+class ONNXUnaryOpLoweringToTOSA : public OpConversionPattern<ONNXOpT> {
+public:
+  using OpConversionPattern<ONNXOpT>::OpConversionPattern;
+  using OpAdaptor = typename ONNXOpT::Adaptor;
+  LogicalResult matchAndRewrite(ONNXOpT op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    rewriter.replaceOpWithNewOp<TOSAOpT>(op, op.getType(), adaptor.X());
+    return success();
+  }
+};
+
+class ONNXReluOpLoweringToTOSA : public OpConversionPattern<ONNXReluOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult matchAndRewrite(ONNXReluOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+
+    // Rescale the input for quantized types. TBD
+    // Maps to tosa.clamp which has both int and fp limits.
+    Value input = adaptor.X();
```

Comment:
We should have a `notifyMatchFailure` for quantized types.

Reply:
Found that a bit strange as well. Maybe a clear comment is better after all.

Reply:
Removed duplicate comment with change request.